tools/c7n_tencentcloud/c7n_tencentcloud/resources/clb.py

```diff
@@ -49,3 +52,115 @@ def augment(self, resources_param):
+                        },
+                        {
+                            "Name": "vpcId",
+                            "Value": metadata_pair[1]
+                        }
+                    ]
+                })
+        return {
+            "Namespace": namespace,
+            "MetricName": self.metric_name,
+            "Period": self.period,
+            "StartTime": self.start_time,
+            "EndTime": self.end_time,
+            "Instances": dimensions
+        }
+
+    def get_metrics_data_point(self, resources, namespace):
+        """
+        yield data_point
+        data_point is a dict which format is the same as DataPoint:
+        {
+            "Dimensions": {
+                "Name": "xxx",
+                "Value": "yyy"
+            }
+            "Timestamps": [int]
+            "Values": [float]
+        }
+        """
+        cli = self.get_client()
+        for batch in chunks(resources, self.batch_size):
+            params = self._get_request_params(batch, namespace)
+            resp = cli.execute_query("GetMonitorData", params)
+            data_points = jmespath.search("Response.DataPoints[]", resp)
+            for point in data_points:
+                yield point
+
+    def _process_open_clbs(self, resources):
+        """
+        resources is a dict which format as below:
+        {
+            key: clb_metadata
+        }
+        """
+        matched_resources = []
+        for point in self.get_metrics_data_point(resources.values(), "QCE/LB_PUBLIC"):
+            if self.match(point):
+                key = point["Dimensions"][0]["Value"]
+                matched_resources.append(resources[key])
+        return matched_resources
+
+    def _process_internal_clbs(self, resources):
+        """
+        resources is a dict which format as below:
+        {
+            key: clb_metadata
+        }
+        """
+        matched_resources = []
+        for point in self.get_metrics_data_point(resources.values(), "QCE/LB_PRIVATE"):
+            if self.match(point):
+                key = f"{point['Dimensions'][1]['Value']}:{point['Dimensions'][0]['Value']}"
+                matched_resources.append(resources[key])
+        return matched_resources
+
+    def process(self, resources, event=None):
+        """process"""
+        print(f"start to process: {len(resources)}, batch_size: {self.batch_size}")
+        # separate resources by LoadBalancerType
+        open_clbs = {}
+        internal_clbs = {}
+        for rs in resources:
+            if rs["LoadBalancerType"] == "OPEN":
+                # for OPEN CLB, just use internet ip as key
+                key = rs['LoadBalancerVips'][0]
+                open_clbs[key] = rs
+            if rs["LoadBalancerType"] == "INTERNAL":
+                for ip in rs["LoadBalancerVips"]:
+                    # for INTERNAL CLB, use the intra ip to generate key
+                    if ipaddress.ip_address(ip.strip()).is_private:
```

Comment:
if its not an IP address what would it be, or is it a guard against transitory state?

Reply:
It is an IP address type.
What we do here is: we just get the metrics data for intra ip and do the filter process.
(in some special case, tencentcloud allows to bind a public ip to the "INTERNAL" CLB)